### PR TITLE
[WIP] Fix for Export and TabHistory

### LINKED ISSUE #180: feat: `export FOO=bar` should modify the shell process's env
this doesn't work - should it?
```
> export MSG=hello
> echo $MSG
```

### LINKED ISSUE #1299: tab context is global - should be local?
Command history and the working directory are not unique to a tab - if you open two tabs, the history of commands executed in one tab is available in the other. Further, changing the working directory in one tab also changes it in the other. I think this can get confusing and localizing the tab context might be more in line with user intuition.

per tab state feature list:
- [ ] history
- [x] cwd
- [x] env

### PR DESCRIPTION
**Pull request by ruanxianzhi**
Fixes #180 
Fixes #1299 

- [x] Export a shell variable as environment variable
- [x] Make sure the exported variable only stay in the current session and tab
- [x] Make history tab-wise
- [x] Add tests

#### Please confirm that your PR fulfills these requirements
- [ ] Multiple commits are squashed into one commit.
- [ ] The commit message follows [Conventional Commits](../CONTRIBUTING.md#conventional-commits), which allows us to autogenerate release notes; e.g. `fix(plugins/plugin-k8s): fixed annoying bugs`
- [ ] All npm dependencies are pinned.

### CODE REVIEW

File: plugins/plugin-bash-like/src/lib/cmds/export.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (starpit) [inline plugins/plugin-bash-like/src/lib/cmds/export.ts]: please add a copyright notice at the top

File: plugins/plugin-bash-like/src/lib/cmds/export.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    // escape quoted equal sign
+  const arr = toBeParsed.split('=')
+
+  curDic[arr[0]] = arr[1]
+
+  storage[tabId] = curDic
+  sessionStore().setItem(key, JSON.stringify(storage))
+  return true
+}
+
```
Comment (starpit) [inline plugins/plugin-bash-like/src/lib/cmds/export.ts]:
```
const usage = {
  command: 'export',
  docs: 'Export a variable or function to the environment of all the child processes running in the current shell'
}
```

File: plugins/plugin-bash-like/src/lib/cmds/export.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Register command handlers
+ *
+ */
+export default (commandTree: CommandRegistrar) => {
+  commandTree.listen('/export', exportCommand, { usage: null, noAuthOk: true, requiresLocal: true })
```
Comment (starpit) [inline plugins/plugin-bash-like/src/lib/cmds/export.ts]: use the above usage mode; and eliminate requiresLocal: true

**Comment by ruanxianzhi** [comment]
Will open two new PR regarding these two issues

**Pull request closed without merging**